### PR TITLE
Refactor test model

### PR DESCRIPTION
**Pull request by swimdi**
### Ticket
#269

### Problem description
The `compile_and_run` fixture will run model with torch_ttnn backend if set torch_ttnn property, but for the training mode it needs to append fake loss function and do the backward, which may differ from models, so needs a method to deal this part.

### What's changed
Implement `ModelTester` class, which handle the `test_model` method, and which do the forward for eval mode and do both forward & backward for training mode, and do the compilation if true the `as_ttnn` arg.`test_model` method also handle the model & input adjustment from different mode.

After this refactor, the testcase under `tests/models` should inherit `ModelTester` class and implement the `_load_model` & `_load_input` method. If other method like `append_fake_loss_function` is not appropriate for current test, can also override it.

So the the testcase under `tests/models` can just use the inherited `ModelTester.test_model()` to run the forward(& backward), and the `compile_and_run` can call it, too.

I want to change the usage from `record_property("torch_ttnn", (model, input, output)` to `record_property("torch_ttnn", (model_test, model_test.test_model()))`, but this change will impact all the `tests/model/` testcase, and I just want to change it all after this PR(i.e. agree usage changed by this refactor) merged, so currently I just adapt `ModelTester` with `tests/models/mnist/mnist.py` and keep the old usage work(so I will remove the old record_property usage after all `tests/model/` adapt the `ModelTester` class)

### CODE REVIEW

**Review by ayerofieiev-tt — APPROVED** [review]
I like it a lot. Thank you!

**Review by kevinwuTT — APPROVED** [review]
I think this is good. Looks like supporting other model input types, like from Tokenizer, might need a few more changes, but that can be in a new PR.

File: tests/models/mnist/test_mnist.py [hunk 1]
```diff
@@ -34,67 +35,28 @@ def forward(self, x):
...
-    m = m.to(torch.bfloat16)
-    m.eval()
-
-    test_input, _ = next(iter(dataloader))
-    test_input = test_input.to(torch.bfloat16)
-    with torch.no_grad():
-        outputs = m(test_input)
-
-    record_property("torch_ttnn", (m, test_input, outputs))
+    record_property("torch_ttnn", (None, None, None))
```
[the remark this answers is not in the document]
Comment (swimdi) [reply tests/models/mnist/test_mnist.py:61]: Yes, will removed after migrate all the tests

File: tests/utils.py [hunk 2]
```diff
@@ -1,5 +1,150 @@
...
+            inputs.requires_grad_(True)
+        else:
+            raise ValueError(f"Current inputs type is not supported: {type(inputs)}")
+        return inputs
+
+    def set_inputs_eval(self, inputs):
+        if type(inputs) == torch.Tensor:
+            inputs = inputs.to(torch.bfloat16)
+        else:
+            raise ValueError(f"Current inputs type is not supported: {type(inputs)}")
```
[the remark this answers is not in the document]
Comment (swimdi) [reply tests/utils.py:56]: If the type is hard to detect or too special(like it is set from tokenizer of _load_inputs stage), I suppose to let that test override set_inputs_*
